src/components/Kanban/Kanban.tsx

```diff
@@ -0,0 +1,272 @@
+                      <KanbanCard
+                        task={task}
+                        taskIndex={index}
+                        provided={provided}
+                        deleteTask={taskIndex => deleteTask(taskIndex)}
+                        updateTaskName={(taskIndex, name) => updateTaskName(taskIndex, name)}
+                      />
+                    ))}
+                    {provided.placeholder}
+                  </div>
+                  <div className="w-full">
+                    {!taskAddMode ? (
+                      <button
+                        className="flex w-full flex-row items-center gap-1 rounded-md px-1 text-left hover:bg-gray-200 dark:hover:bg-gray-600"
+                        onClick={() => setTaskAddMode(true)}
+                      >
+                        <BsPlus className="h-6 w-6" />
+                        <span className="align-middle">Add Task</span>
+                      </button>
+                    ) : (
+                      <form onSubmit={e => onFormSubmit(e)}>
+                        <div className="px-1">
+                          <input
+                            autoFocus
+                            value={taskInputValue}
+                            onChange={event => {
+                              setTaskInputValue(event.target.value);
+                            }}
+                            placeholder="Enter a task name..."
+                            className="mb-2 w-full overflow-hidden rounded-sm border border-gray-300 p-1 dark:border-gray-500 dark:bg-gray-700"
+                          />
+                          <div className="flex w-full flex-row gap-1">
+                            <button
+                              ref={addTaskButtonRef}
+                              className="rounded-md bg-blue-600 px-2 py-0.5 text-white hover:bg-blue-700"
+                            >
+                              Add Task
+                            </button>
+                            <button
+                              className="rounded-md px-2 py-0.5 hover:bg-gray-300 dark:hover:bg-gray-600"
+                              onClick={() => setTaskInputValue("")}
+                            >
+                              Cancel
+                            </button>
+                          </div>
+                        </div>
+                      </form>
+                    )}
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        );
+      }}
+    </Droppable>
+  );
+};
+
+export const Kanban = ({}) => {
+  const { setIsKanbanToggled } = useToggleKanban();
+  const { board, setColumns } = useKanban();
+  const isDesktop = useMediaQuery("(min-width: 641px)");
+
+  const addTask = (columnIndex: number, taskName: string) => {
+    let columns = board.columns;
+    columns[columnIndex].tasks.push({ id: v4(), name: taskName });
```

Comment:
Would it possibly make sense to double check if there is an existing task in the kanboard with the same name?

Reply:
If we want names to be unique, we could add that. Totally up to you.
At the moment it could make sense but if we ever added more complex features like descriptions, you could have cards with the same name but different descriptions inside, so not sure if we want to enforce this limit.

Reply:
Thought about it a bit more and imo enforcing this limit doesn't make too much sense and we should just let the user do whatever they want.